integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -135,7 +136,24 @@ private[sql] case class CarbonProjectForUpdateCommand(
           else {
             Dataset.ofRows(sparkSession, plan)
           }
-
+          if (CarbonProperties.getValidateUpdateKeyValueMapping) {
+            // If more than one value present for the update key, should fail the update
+            val ds = dataSet.select(CarbonCommonConstants.CARBON_IMPLICIT_COLUMN_TUPLEID)
+              .groupBy(CarbonCommonConstants.CARBON_IMPLICIT_COLUMN_TUPLEID)
+              .count()
+              .select("count")
+              .filter(col("count") > lit(1))
+              .limit(1)
+              .collect()
```

Comment:
@zzcclp @ajantha-bhat  One suggestion is to try to move this check to the join subquery in the update operation itself, then we do not need to do this check separately.

Reply:
@jackylk : yes, it is now in the same place as you are suggesting.
line 135, we are making `Dataset.ofRows(sparkSession, plan)`, on top of this only I am collecting tupleId.